idb/postgres/postgres.go

```diff
@@ -268,12 +292,19 @@ func (db *IndexerDb) CommitBlock(round uint64, timestamp int64, rewardslevel uin
 		return fmt.Errorf("put block_header %v    %#v", err, err)
 	}
 
-	err = tx.Commit()
+	return tx.Commit()
+}
+
+// CommitBlock is part of idb.IndexerDB
+func (db *IndexerDb) CommitBlock(round uint64, timestamp int64, rewardslevel uint64, headerbytes []byte) error {
+	f := func() error {
+		return db.commitBlock(round, timestamp, rewardslevel, headerbytes)
+	}
+	err := txnWithRetry(f)
+
 	db.txrows = nil
 	db.txprows = nil
```

Comment:
Why not put these into `commitBlock` so that you can return txnWithRetry like the other wrapper?

Reply:
I don't want to modify state in a function that might be repeated.